doc/specs/#885 - Terminal Settings Model/Actions Addendum.md

```diff
@@ -0,0 +1,310 @@
+ something like the following:
+
+```c++
+runtimeclass ActionMap
+{
+    ActionAndArgs GetActionByKeyChord(KeyChord keys);
+
+    KeyChord GetKeyBindingForAction(ShortcutAction action);
+    KeyChord GetKeyBindingForAction(ShortcutAction action, IActionArgs actionArgs);
+
+    IMapView<String, Command> NameMap { get; };
+
+    // Future Considerations:
+    // - [#6899]: Action IDs --> GetActionByID()
+}
+```
+
+The getters will return null if a matching action or key chord is not found. Since iterable commands need to be expanded at in TerminalApp, we'll just expose `NameMap`, then let TerminalApp perform the expansion as they do now. Internally, we can store the actions as follows:
+
+```c++
+std::map<KeyChord, InternalActionID> _KeyMap;
+std::map<InternalActionID, Command> _ActionMap;
+```
+
+`InternalActionID` will be a hash of `ActionAndArgs` such that two `ActionAndArgs` with the same `ShortcutAction` and `IActionArgs` output the same hash value.
+
+`GetActionByKeyChord` will use `_KeyMap` to find the `InternalActionID`, then use the `_ActionMap` to find the bound `Command`.
+`GetKeyBindingForAction` will hash the provided `ActionAndArgs` (constructed by the given parameters) and check `_ActionMap` for the given `InternalActionID`.
+
+`NameMap` will need to ensure every action in `_ActionMap` is added to the output name map if it has an associated name. This is done by simply iterating over `_ActionMap`. Nested commands must be added separately because they cannot be hashed.
+
+`ActionMap` will have an `AddAction(Command cmd)` that will update the internal state whenever a command is registered. If the given command is valid, we will check for collisions and resolve them. Otherwise, we will consider this an "unbound" action and update the internal state normally. It is important that we don't handle "unbound" actions differently because this ensures that we are explicitly unbinding a key chord.
+
+### Step 3: Settings UI needs
+
+After the former two steps are completed, the new representation of actions in the settings model is now on-par with
+ what we have today. In order to bind these new actions to the Settings UI, we need the following:
+
+1. Exposing the maps
+    - `ActionMap::KeyBindings` and `ActionMap::Commands` may need to be added to pass the full list of actions to the Settings UI.
+    - In doing this, we can already update the Settings UI to include a better view of our actions.
+2. Creating a copy of the settings model
+    - The Settings UI operates by binding the XAML controls to a copy of the settings model.
+    - Copying the `ActionMap` is fairly simple. Just copy the internal state and ensure that `Command::Copy` is called such that no reference to the original WinRT objects persist. Since we are using `InternalActionID`, we will not have to worry about multiple `Command` references existing within the same `ActionMap`.

```

Comment:
Added a short segment at the end of part 3 (Modifying the `Command`s). I don't feel like we should spec the view model part out too much here since the spec is intended to focus more on the settings model changes, but it is worth a callout.